### DISCUSSION TITLE
How to reduce the quantity item every time new form created

### DISCUSSION THREAD

**Discussion by faeyzanramzan** [filamentphp/filament#10692] [Help]
I'm trying to make a booking system where user can reserved an item. But I'm having problem with the quantity of the item, I want to reduce the item from the database. where do I need to code if I want to reduce the quantity

Here is my InvoiceResouces.php
```
public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Group::make()
                ->schema([
                    Card::make()
                    ->schema([
                        TextInput::make('invoice_number')->required(),
                        DatePicker::make('invoice_date')->default(now())->required()
                    ])->columns([
                        'sm'=>2,
                    ]),
                    Card::make()
                    ->schema([
                        Placeholder::make('Products'),
                        Repeater::make('invoiceItems')
                        ->relationship()
                        ->schema([
                            Select::make('product_id')
                            ->label('Product')
                            ->options(Product::query()->pluck('name','id')),
                            TextInput::make('quantityItems')

                        ])
                    ])
                    
                ])->columnSpan('full')
            ]);
    }
```

Here is my ProductResource.php
```
public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Card::make()
                ->schema([
                    TextInput::make('serial_no')->required(),
                    TextInput::make('name')->required(),

                    Select::make('brand_id')
                    ->relationship('brand','name'),
                    Select::make('type_id')
                    ->relationship('type','name'),

                    TextInput::make('quantity')->required()
                ])
               

            ]);
    }
```
I try to search online, it can use `decrement ()` but I don't know where to put it. Thank you for helping me

**Comment by mahmudz** [comment]
Did you mean, every time new invoice is created, the product quantity will be reduced on products table?

**Reply by faeyzanramzan** [reply]
yes, correct

**Reply by mahmudz** [reply]
In this case you can use Eloquent events in Invoice model, so whenever an invoice is created, update the quantity of those products associated with the invoice.

https://laravel.com/docs/10.x/eloquent#events-using-closures

**Reply by faeyzanramzan** [reply]
Example like this?
```
 static::updating(function (Invoice $invoice) {
            //code for reduced item
        });
```

**Reply by mahmudz** [reply]
Yes.

I think you'll the created one instead of updated if you want to change the quantity on new invoice creation. 
Make sure to put this in booted method as shown in the documentation

**Reply by faeyzanramzan** [reply]
so the code to reduced item, i only use `decrement ()` ?

**Reply by mahmudz** [reply]
https://laravel.com/docs/10.x/queries#increment-and-decrement

Maybe this can help you.

**Reply by kombiahrk** [reply]
@faeyzanramzan 

in your InvoiceResource you can this 

`->mutateRelationshipDataBeforeCreateUsing(function (array $data, $record, Forms\Get $get): array 
{
     //code for item reducing
}`

or in CreateInvoice of Particular Resource you can use this method

`protected function afterCreate(): void
{
    //code for item reducing
}`
